### PR TITLE
Tauffer

### LINKED ISSUE #15: all times should be in seconds after session_start_time

### PR DESCRIPTION
**Pull request by luiztauffer**
fix #15

### CODE REVIEW

**Comment by luiztauffer** [comment]
@bendichter it's a small conflict, we both renamed the `name` attribute for Units in different commits. I'm leaving it up to you to choose.

File: src/buffalonwb/add_units.py [hunk 1]
```diff
@@ -83,8 +84,8 @@ def add_units(nwbfile, nex_file_name, include_waveforms=False):
         else:
             kwargs.update(num_samples=var_header['Count'])
         nwbfile.add_unit(electrodes=(int(var_header['Name'][3:-2]) - 1,),
-                         name=var_header['Name'],
-                         spike_times=var['Timestamps'],
+                         unit_name=var_header['Name'],
```
Comment (bendichter) [inline src/buffalonwb/add_units.py]:
```suggestion
                         label=var_header['Name'],
```

File: src/buffalonwb/metafile.yml [hunk 2]
```diff
@@ -35,13 +35,6 @@ NWBFile:
   session_id: ADDME
   related_publications:
   session_start_time: 2017-05-04 11:23:13
-  slices:
   source_script: https://github.com/ben-dichter-consulting/buffalo-lab-data-to-nwb
   source_script_file_name: conversion_module.py
-  stimulus_notes:
```
Comment (bendichter) [inline src/buffalonwb/metafile.yml:41]: why were these removed? I think we should still have optional fields in the YAML file.

File: src/buffalonwb/metafile.yml [hunk 3]
```diff
@@ -2,25 +2,25 @@ Behavior:
   EyeTracking:
     name: EyeTracking
     spatial_series:
-      data: true
-      description: eye tracking
-      name: SpatialSeries
-      reference_frame: ''
+      - data: true
```
Comment (bendichter) [inline src/buffalonwb/metafile.yml]: What is the purpose of the `data: true` statement

File: src/buffalonwb/add_behavior.py [hunk 4]
```diff
@@ -14,7 +18,8 @@ def add_behavior(nwbfile, behavior_file, metadata_behavior):
     )
 
     # Player Position
-    pos = Position(name='Position')
+    meta_pos = metadata_behavior['Position']
+    pos = Position(name=meta_pos['name'])
```
Comment (bendichter) [inline src/buffalonwb/add_behavior.py:22]: we should assign description here if defined. Would `pos = Position(**meta_pos)` work?

File: src/buffalonwb/metafile.yml (hunk 2 above)
Comment (luiztauffer) [reply src/buffalonwb/metafile.yml:41]: any field can be included or removed! I removed those ones without info, but if the user add them, they'll be used automatically by the GUI/conversion

File: src/buffalonwb/metafile.yml (hunk 3 above)
Comment (luiztauffer) [reply src/buffalonwb/metafile.yml]: data is one of the things we can't add through metadata, so the idea is to give an optional control to the user whether to include specific data or not. I think it could be useful with, say, multiple TimeSeries, and for a particular case the user decides to add just some of them

File: src/buffalonwb/add_behavior.py (hunk 4 above)
Comment (luiztauffer) [reply src/buffalonwb/add_behavior.py:22]: [Position](https://pynwb.readthedocs.io/en/stable/pynwb.behavior.html#pynwb.behavior.Position) does not have description.
`pos = Position(**meta_pos)` would not work because what's on `spatial_series` field is metadata about the spatial_series to be constructed, not SpatialSeries objects

File: src/buffalonwb/metafile.yml (hunk 3 above)
Comment (bendichter) [reply src/buffalonwb/metafile.yml]: But data is a required field of TimeSeries, so they shouldn't be able to set this to false because that would result in an invalid NWB file. I don't see how this boolean flag helps users pick between datasets

File: src/buffalonwb/metafile.yml (hunk 2 above)
Comment (bendichter) [reply src/buffalonwb/metafile.yml:41]: Can we include all of the optional fields by default?

Comment (luiztauffer) [reply src/buffalonwb/metafile.yml:41]: If you want so, of course! It makes the yaml file less readable though, packed with not used info.

File: src/buffalonwb/metafile.yml (hunk 3 above)
Comment (luiztauffer) [reply src/buffalonwb/metafile.yml]: The way I pictured it was being used as a boolean control before creating or adding the objects to nwbfile:
```pyhton
if meta['Behavior']['Position']['spatial_series'][i]['data'] == True:
	make spatial_series object
	add it to nwbfile
```

looking at it now I agree it feels counter-intuitive that this control of add/don't add will be done by the `data` field. Anyway, it's just an idea of use for the `data` field on metadata, we can as well just remove it from GUI/metadata.

Comment (luiztauffer) [reply src/buffalonwb/metafile.yml]: I could make the Series group toggle active/inactive in the GUI with the checking/unchecking of the data checkbox, that could be a visually intuitive way to see what's going on.

Comment (bendichter) [reply src/buffalonwb/metafile.yml]: Yeah, I agree that sounds visually intuitive. I don't know if we should associate that with the `data` boolean flag though.

Comment (luiztauffer) [reply src/buffalonwb/metafile.yml]: I am removing the `data` field from metafiles and GUI. This is a general conversion point, not specific to this repo alone.
